fix(agentic-ai): Ignore blank assistant text to avoid IllegalArgumentException

nourzakhama2003 opened the pull request:
##  summary 
Fix an intermittent incident where the Generic AI Agent connector throws IllegalArgumentException("Text cannot be null or empty") when the AI assistant returns blank or whitespace-only text. This change drops blank assistant text instead of constructing [TextContent] from it.


## Description

Problem :
Symptom: intermittent incidents labeled "Text cannot be null or empty" (issue #5226).
Root cause: [ChatMessageConverterImpl] created [TextContent] whenever [aiMessage.text() != null].
 The [TextContent] record constructor throws for blank or whitespace-only text, so certain AI responses caused an exception.
Solution :
Only create [TextContent] when the assistant text is non-blank:
[if (StringUtils.isNotBlank(aiMessage.text())) { ... }]
Added unit test: [toAssistantMessage_ignoresBlankText()] in [ChatMessageConverterTest].

Files changed :
Modified: [ChatMessageConverterImpl.java]
Modified: [ChatMessageConverterTest.java]

Tests & Verification 💯 
New unit test added and verified locally.
Module tests: — 987 tests passed, 0 failures (local verification).

<img width="1124" height="415" alt="Capture d&#39;écran 2025-12-24 040925" src="https://github.com/user-attachments/assets/203a674e-4802-4838-9396-c8458fb87f92" />


Low-risk: behavior only ignores blank assistant text (previously threw an exception). No public API changes.

##  Checklist 
 Implement fix with focused change
 Add unit test reproducing the issue
 Run module tests locally (all pass)
 Request review from agentic-ai maintainers

## Related issues
Potentially related to #5226


## Checklist

- [x] PR has a **milestone** or the `no milestone` label.
- [x] Backport labels are added if these code changes should be backported. No backport label is added to the latest release, as this branch will be rebased onto main before the next release.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/connectors?pullRequest=5976) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/connectors?pullRequest=5976) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/connectors?pullRequest=5976) it.</sub>

nourzakhama2003 commented:
Fixed Windows test issues by replacing getResource().getFile() with toURI() and adjusted method signatures where needed.                            Small CI workflow improvement to upload surefire XML artifacts so test reporting finds results reliably (see TEST_FEATURE_BRANCH.yml).

nourzakhama2003 commented:
> @nourzakhama2003 thanks for your contribution!
> 
> Could you please trim down the changes on the PR to only address the assistant text issue & remove the icon and CI changes? Happy to keep the test changes regarding Windows compatibility, thanks for addressing those!

Hi @maff / maintainers — update: I reset the branch to keep only the single fix commit (1afe1c58f) that ignores blank assistant text. I also reverted the icon & CI workflow edits; the branch now only changes the converter + Windows test fixes.
